addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -991,26 +991,57 @@ export class OdooEditor extends EventTarget {
     }
 
     updateColorpickerLabels(params = {}) {
-        const foreColor = params.foreColor || rgbToHex(document.queryCommandValue('foreColor'));
+        function hexFromColor(color) {
+            if (color && color.includes('-gradient(')) {
+                color = color.match(/gradient(.*)/)[0];
+                let r = 0, g = 0, b = 0, count = 0;
+                for (const entry of color.matchAll(/rgba?\(([0-9]+), ([0-9]+), ([0-9]+)[, [0-9]*]?\)/g)) {
```

Comment:
This one is slightly different because it obtains all colors in a single string.
I now recovered the regex from colorpicker.js, removed the `^` and `$` and added the `/g` to make it global.